### PR TITLE
Migrations Generator with invalids attributes

### PR DESCRIPTION
**Pull request by paulociecomp**
Add warn when generate model and migrations with attibutes with invalids characters and like Model type:string because "type" is a keyword.

### CODE REVIEW

**Comment by egilburg** [comment]
`type` is not a hardcoded keyword, it's the default inheritnace column but users can choose a different column instead and treat `type` like any other column.

http://api.rubyonrails.org/classes/ActiveRecord/ModelSchema/ClassMethods.html#method-i-inheritance_column

**Comment by paulociecomp** [comment]
@egilburg thanks for the feedback. I had not thought about that. I did some changes. I don't know if this feature will be accepts any way. Waiting more feedbacks.

**Comment by rafaelfranca** [comment]
If we are doing this how would you create a table that works with STI using generators?

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+*   Display warning when generate model with invalids attributes
+
```
Comment (shunsukeaida) [inline activerecord/CHANGELOG.md]: when generating?

**Comment by shunsukeaida** [comment]
I was about to send a similar PR (but too busy) , but how about referring to `type` as a public const from  `ActiveRecord::ModelSchema` or other modules?
```ruby
module ActiveRecord
  module Inheritance
    extend ActiveSupport::Concern
    DEFAULT_INHERITANCE_COLUMN_NAME = "type"
  end
end
```

**Comment by paulociecomp** [comment]
@rafaelfranca it just display a warning. We can use STI normally.

File: activerecord/CHANGELOG.md (hunk 1 above)
Comment (paulociecomp) [inline activerecord/CHANGELOG.md]: @shunsukeaida thanks.

**Comment by paulociecomp** [comment]
@rafaelfranca @shunsukeaida now i get it. Sorry. I didn't realize it. Do you have any suggestions?

**Comment by shunsukeaida** [comment]
I'd wonder if we should check it on migration, when we can do upon the model creation. Maybe less so, considering that [`Generators::ModelHelpers`](https://github.com/rails/rails/blob/master/railties/lib/rails/generators/model_helpers.rb) already has a check for a plural name and we can just expand this section a bit more to have a couple more checks.  My two cents.

**Comment by matthewd** [comment]
I'm :-1: on this in general.

I understand it would be frustrating if you've just created a model without realising that `type` is magical. But if we wanted to solve this, would the generator really be the right layer to do it?

And then there's the fact that the warning makes it feel like you're doing something "wrong", even when you actually intend to make an inheritance column.

**Comment by rafaelfranca** [comment]
Agree with @matthewd about "fell like you're doing something wrong"

**Comment by paulociecomp** [comment]
@matthewd @rafaelfranca makes sense. Thanks.

**Pull request closed without merging**